apps/src/blockly/cdoBlocklyWrapper.js

```diff
@@ -220,12 +224,28 @@ function initializeBlocklyWrapper(blocklyInstance) {
   };
 
   blocklyWrapper.getWorkspaceCode = function (opt_showHidden) {
-    const code = Blockly.Generator.blockSpaceToCode(
-      'JavaScript',
-      null,
-      !!opt_showHidden
-    );
-    return strip(code);
+    let code = '';
+    try {
+      code = Blockly.Generator.blockSpaceToCode(
+        'JavaScript',
+        null,
+        !!opt_showHidden
+      );
+      code = strip(code);
+      getStore().dispatch(setFailedToGenerateCode(false));
+    } catch (e) {
+      // We only want to log the error once per failure since getWorkspaceCode
+      // gets called many times and the error will be the same every time.
+      if (!getStore().getState().blockly.failedToGenerateCode) {
+        getStore().dispatch(setFailedToGenerateCode(true));
+        MetricsReporter.logError({
+          event: 'CDO_BLOCKLY_GET_CODE_ERROR',
```

Comment:
Nit: I usually recommend avoiding literal and make them constants
```
    const ERROR_EVENT_NAME = 'CDO_BLOCKLY_GET_CODE_ERROR';
        MetricsReporter.logError({
      event: ERROR_EVENT_NAME,
      errorMessage: e.message,
      stackTrace: e.stack,
    });
    ```
    
By keeping this style, one day you will run into a value you will need to change and remember this because it will make it easier :D It has happened to me a couple of times. 

I prefer all constant for metric events into a single file. They all reference it, that way, you have a way to know which events are being logged too.

Reply:
Seems fine to me!